mock-core-configs/etc/mock/templates/amazonlinux-2023.tpl

```diff
@@ -0,0 +1,53 @@
+config_opts['root'] = 'amazonlinux-2023-{{ target_arch }}'
+config_opts['chroot_setup_cmd'] = 'install --allowerasing @buildsys-build dnf'
```

Comment:
uhh, what's `--allowerasing` here for? Also `@buildsys-build` doesn't exist. And do you really need `dnf` in there too? We don't use `dnf` inside the mock root...

Reply:
I don't seem to need `--allowerasing` nor `dnf` there - these are likely some fun artifacts from elsewhere that came across copy pasta.

The `@buildsys-build` is grabbing the group from comps, which is:
```
  <group>
    <id>buildsys</id>
    <name>buildsys-build</name>
    <description>The base set of packages for a mock chroot</description>
    <default>false</default>
    <uservisible>true</uservisible>
    <packagelist>
<packagereq type="mandatory">system-release</packagereq>
      <packagereq type="mandatory">bash</packagereq>
      <packagereq type="mandatory">bzip2</packagereq>
      <packagereq type="mandatory">coreutils</packagereq>
      <packagereq type="mandatory">cpio</packagereq>
      <packagereq type="mandatory">diffutils</packagereq>
      <packagereq type="mandatory">file</packagereq>
      <packagereq type="mandatory">findutils</packagereq>
      <packagereq type="mandatory">gawk</packagereq>
      <packagereq type="mandatory">gcc-c++</packagereq>
      <packagereq type="mandatory">gcc</packagereq>
      <packagereq type="mandatory">git</packagereq>
      <packagereq type="mandatory">glibc-all-langpacks</packagereq>
      <packagereq type="mandatory">glibc-locale-source</packagereq>
      <packagereq type="mandatory">grep</packagereq>
      <packagereq type="mandatory">gzip</packagereq>
      <packagereq type="mandatory">info</packagereq>
      <packagereq type="mandatory">kernel-devel</packagereq>
      <packagereq type="mandatory">kernel</packagereq>
      <packagereq type="mandatory">make</packagereq>
      <packagereq type="mandatory">patch</packagereq>
      <packagereq type="mandatory">python-rpm-macros</packagereq>
      <packagereq type="mandatory">python3-rpm-macros</packagereq>
      <packagereq type="mandatory">system-rpm-config</packagereq>
      <packagereq type="mandatory">rpm-build</packagereq>
      <packagereq type="mandatory">rpmdevtools</packagereq>
      <packagereq type="mandatory">sed</packagereq>
      <packagereq type="mandatory">shadow-utils</packagereq>
      <packagereq type="mandatory">tar</packagereq>
      <packagereq type="mandatory">unzip</packagereq>
      <packagereq type="mandatory">util-linux</packagereq>
      <packagereq type="mandatory">wget</packagereq>
      <packagereq type="mandatory">which</packagereq>
      <packagereq type="mandatory">xz</packagereq>
    </packagelist>
  </group>
```

Reply:
Ah, so it should be `@buildsys` (I think you might have made a mistake when you defined your comps group, the _ID_ should be `buildsys-build`, and the _name_ can be a title of the group `Build system environment`).

Reply:
(Also, as an aside, your `@buildsys` group is quite a bit greedier than Fedora's)

Reply:
That is possible... I'll check what we're doing on AL2 and our koji stack. It's possible we have very much inherited this over the past decade and barely/never used it.

Reply:
Here's what Fedora's is, as a reference (taken from `f39-comps.xml.in`):

```xml
  <group>
    <id>buildsys-build</id>
    <_name>Buildsystem building group</_name>
    <_description/>
    <default>false</default>
    <uservisible>false</uservisible>
    <packagelist>
      <packagereq type="mandatory">bash</packagereq>
      <packagereq type="mandatory">bzip2</packagereq>
      <packagereq type="mandatory">coreutils</packagereq>
      <packagereq type="mandatory">cpio</packagereq>
      <packagereq type="mandatory">diffutils</packagereq>
      <packagereq type="mandatory">fedora-release-common</packagereq>
      <packagereq type="mandatory">findutils</packagereq>
      <packagereq type="mandatory">gawk</packagereq>
      <packagereq type="mandatory">glibc-minimal-langpack</packagereq>
      <packagereq type="mandatory">grep</packagereq>
      <packagereq type="mandatory">gzip</packagereq>
      <packagereq type="mandatory">info</packagereq>
      <packagereq type="mandatory">patch</packagereq>
      <packagereq type="mandatory">redhat-rpm-config</packagereq>
      <packagereq type="mandatory">rpm-build</packagereq>
      <packagereq type="mandatory">sed</packagereq>
      <packagereq type="mandatory">shadow-utils</packagereq>
      <packagereq type="mandatory">tar</packagereq>
      <packagereq type="mandatory">unzip</packagereq>
      <packagereq type="mandatory">util-linux</packagereq>
      <packagereq type="mandatory">which</packagereq>
      <packagereq type="mandatory">xz</packagereq>
    </packagelist>
  </group>
```

Reply:
Maybe instead of using `@buildsys`, we just enumerate a list of packages for now, because I don't think we want mock chroots getting kernels installed and stuff. That's too much and can make buildroot setup very slow.

Reply:
At some point in the past I did sync what we have in koji to the comps groups, but it's possible that it's out of sync.... will confirm.

Reply:
Importantly, I think in Koji, this group is called `@build`, and we sync this and export it to comps as `@buildsys-build`.

You can see it when you point `koji(1)` to Fedora's instance and do `koji show-groups f39-build`.